Pocket/pocket-ios
# Premium Purchase - Hooray! Screen

## Summary
* This PR presents the Hooray! screen at the end of a successful purchase, either from Search and from Settings.
* It also adds a separate image asset for dark mode, for the same screen.

## Implementation Details
* Add `premium.hooray.dark` asset and rename existing asset to `premium.hooray.light`
* Update `SettingsView`, `AccountViewModel`, `SearchView`, `SearchViewModel`: add logic to present the Hooray! screen after successful purchase
* Update `PremiumUpgradeSuccessScreen` use different images for light and dark mode

## Test Steps
### Pre-requisites
- Configure the Xcode testing environment as described in #421 
- Set the property `shouldAllowUpgrade` in `PremiumUpgradeView` to `true`

### Testing steps
* Checkout this branch then build/run
* Go to Settings > Go Premium and purchase a subscription, then tap "OK" on the "You're all set!" alert
* Make sure the Hooray! screen appears and it's dismissible from either the "Back to Pocket" button or the X in the top left

### Optional steps
- delete any transaction in the StoreKit testing environment in Xcode (or just use another device)
- Repeat the above steps from Search > All Items > "Get Pocket Premium"

## PR Checklist:

## Animations

iPhone Dark - Settings | iPad Light - Search
-- | --
<img height=660 width=300 src=https://user-images.githubusercontent.com/34376330/225172586-7711fc24-250f-456c-b82d-f994c9836b1a.gif> | <img height=660 src=https://user-images.githubusercontent.com/34376330/225172607-01391939-5ab6-49d9-bf50-b41d3fe358a7.gif>
